### ISSUE TITLE
null string handling difference between NET Core and NET 5.0

### ISSUE THREAD

**Issue by gewarren** [dotnet/docs#22827]
---
**Issue moved from MicrosoftDocs/feedback#3406**
- Please respond to @Ray-Wynn.

---

_From @Ray-Wynn on Friday, January 22, 2021 5:33:46 AM_

parameter null character as string ("\0") allowed in NET Core but not in NET 5.0 as shown below.
``
static void Main(string[] args)
        {            
            byte[] testBytes = { 0x41, 0x42, 0x43, 0x0, 0x0, 0x0, 0x0, 0x0 }; // ABC\0\0\0\0\0
            string testString = Encoding.ASCII.GetString(testBytes);

            int ns = testString.IndexOf("\0");  // .Net Core 3.x = 3 - .Net 5.0 = 0
            int nc = testString.IndexOf('\0');  // .Net Core 3.x = 3 - .Net 5.0 = 3

            Console.WriteLine(string.Format("testString={0}, IndexOf: null string = {1:d} null character = {2:d}", testString, ns, nc));
            Console.ReadKey();
        }
``

**Comment by gewarren** [comment]
---
**Issue moved from MicrosoftDocs/feedback#3406**
- Please respond to @welcome[bot].

---

_From @welcome[bot] on Friday, January 22, 2021 5:33:48 AM_

Thank you for creating the issue! One of our team members will get back to you shortly with additional information. If this is a product issue, please close this and contact the particular product's support instead (see https://support.microsoft.com/allproducts for the list of support websites).

**Comment by gewarren** [comment]
This is likely due to the switch to use ICU for globalization facilities. https://github.com/MicrosoftDocs/feedback/issues/3406

If desired, you can revert back to using NLS by following the instructions here: https://github.com/MicrosoftDocs/feedback/issues/3406.

I'll close this issue now; please leave a comment if you think there is a different problem.

**Comment by jzs11** [comment]
What? what's the resolution here?

**Comment by Ray-Wynn** [comment]
Hardly worth the effort of reporting. Despite being found in FTDI device driver code. At least FTDI appreciated my heads up on the issue. Microsoft response underwhelming, as usual.

Regards
Ray

**Comment by gewarren** [comment]
Sorry, I don't know what happened to my links of my Feb. 12th comment! Those obviously were not the links I intended to paste in.

I confirmed that this is due to the NLS -> ICU switch on Windows in .NET 5. If you do an ordinal string comparison, your test code will return an index of 3 even on .NET 5:

```
testString.IndexOf("\0", StringComparison.Ordinal)
```

The links I meant to paste in are these:

- [Behavior differences](https://docs.microsoft.com/dotnet/standard/base-types/string-comparison-net-5-plus#behavioral-differences)
- [To revert to the previous behavior](https://docs.microsoft.com/dotnet/standard/globalization-localization/globalization-icu#use-nls-instead-of-icu)
